mat/band.go

```diff
@@ -297,3 +298,36 @@ func (b *BandDense) Trace() float64 {
 	}
 	return tr
 }
+
+// MulVecTo computes B⋅x or Bᵀ⋅x storing the result into dst.
+func (b *BandDense) MulVecTo(dst *VecDense, trans bool, x Vector) {
+	m, n := b.Dims()
+	if trans {
+		m, n = n, m
+	}
+	if x.Len() != n {
+		panic(ErrShape)
+	}
+	dst.reuseAsNonZeroed(m)
+
+	t := blas.NoTrans
+	if trans {
+		t = blas.Trans
+	}
+
+	xMat, _ := untransposeExtract(x)
+	if xVec, ok := xMat.(*VecDense); ok {
+		if dst != xVec {
+			dst.checkOverlap(xVec.mat)
+			blas64.Gbmv(t, 1, b.mat, xVec.mat, 0, dst.mat)
+		} else {
+			var xCopy VecDense
+			xCopy.CloneVec(xVec)
+			blas64.Gbmv(t, 1, b.mat, xCopy.mat, 0, dst.mat)
+		}
+	} else {
+		var xCopy VecDense
+		xCopy.CloneVec(x)
+		blas64.Gbmv(t, 1, b.mat, xCopy.mat, 0, dst.mat)
+	}
```

Comment:
i'm wondering if these would benefit from something like this.
```
		} else {
			xCopy := getWorkspaceVec(n, false)
			xCopy.CloneVec(xVec)
			blas64.Gbmv(t, 1, b.mat, xCopy.mat, 0, dst.mat)
			putWorkspaceVec(xCopy)
		}
	} else {
		xCopy := getWorkspaceVec(n, false)
		xCopy.CloneVec(x)
		blas64.Gbmv(t, 1, b.mat, xCopy.mat, 0, dst.mat)
		putWorkspaceVec(xCopy)
	}
```

Reply:
Done. AFAIU, the more the workspace is used, the more useful it is?

Reply:
Yes. A benchmark might be worthwhile, but that can happen later.